validator/client/propose.go

```diff
@@ -122,31 +122,59 @@ func (v *validator) ProposeBlock(ctx context.Context, slot primitives.Slot, pubK
 	}
 
 	var genericSignedBlock *ethpb.GenericSignedBeaconBlock
-	if blk.Version() >= version.Deneb && !blk.IsBlinded() {
-		signedBlobs := make([]*ethpb.SignedBlobSidecar, len(b.GetDeneb().Blobs))
-		for i, blob := range b.GetDeneb().Blobs {
-			blobSig, err := v.signBlob(ctx, blob, pubKey)
+	if blk.Version() >= version.Deneb {
+		if !blk.IsBlinded() {
+			signedBlobs := make([]*ethpb.SignedBlobSidecar, len(b.GetDeneb().Blobs))
```

Comment:
you should probably extract this into its own function to set generic signed block since it's going above the cognitive complexity

Reply:
[d7b63c5](https://github.com/prysmaticlabs/prysm/pull/12922/commits/d7b63c538fa7d4e6833f3f01df2d1a5c8b1fa494)